### PR TITLE
chore: update to netstandard2.1

### PR DESCRIPTION
**Pull request by stijnmoreels**
Updates the projects from `netstandard2.0` to `netstandard2.1`.

Relates to https://github.com/arcus-azure/arcus/issues/106

### CODE REVIEW

File: src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
   <ItemGroup>
     <PackageReference Include="Guard.Net" Version="1.2.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.8" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj:22]: Changing it like this, feels like a breaking change. But not sure if it technically qualifies as a breaking change for us.

Comment (fgheysels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj:22]: I don't think it is a problem to use 3.1.8 in .NEt 6 projects.   I don't know if you can use 6.0.0 in .NET Core projects ...

Comment (stijnmoreels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj]: Ok, I reverted back to 3.1.8. Should we support both, for now? I mean .NET 6 and .NET Core? Maybe that's a good thing, for now.

Comment (fgheysels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj]: If possible, yes, I think we should support .NET 6 & .NET Core

Comment (stijnmoreels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj]: Ok, sounds like a plan 👍 . We can complete this one now, then.

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj (hunk 1 above)
Comment (fgheysels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj]: Seems like I'm not a code owner or something, as my review is not enough to complete this. 
I can also not assign someone to this PR

Comment (stijnmoreels) [reply src/Arcus.Testing.Logging/Arcus.Testing.Logging.csproj]: Aha, I'll change that.

**Pull request merged**